pipeline_dp/pipeline_operations.py

```diff
@@ -156,8 +157,11 @@ def map_tuple(self, col, fn, stage_name: str = None):
     def map_values(self, col, fn, stage_name: str):
         pass
 
-    def group_by_key(self, col, stage_name: str):
-        pass
+    def group_by_key(self, col, stage_name: str = None):
```

Comment:
We need to have it lazy, d.items() is generator-like object, but for-loop is executed eagerly. What do you think about the following approach

```
def group_by_key(self, ...):
   def group_by_key_generator(...):
         d = defaultdict(list)
         for key, value in col:
             d[key].append(value)
         for item in d.items():
             yield item
   return group_by_key_generator
```

Reply:
I updated it, let me know what you think.
Thanks for catching this I didn't consider that

Reply:
Ah, yeah, thanks, () was missing in "return group_by_key_generator"